### PR TITLE
test: add a test to verify an important contract of setParent

### PR DESCRIPTION
**Pull request by monperrus**

### CODE REVIEW

File: src/test/java/spoon/test/parent/SetParentTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import static spoon.test.parent.ParentContractTest.createReceiverList;
+import static spoon.testing.utils.ModelUtils.createFactory;
+
+// contract: setParent does not modifiy the state of the parent
+@RunWith(Parameterized.class)
+public class SetParentTest<T extends CtVisitable> {
+
+	private static Factory factory = createFactory();
+
+	private static final List<CtType<? extends CtElement>> allInstantiableMetamodelInterfaces = SpoonTestHelpers.getAllInstantiableMetamodelInterfaces();
```
Comment (pvojtechovsky) [inline src/test/java/spoon/test/parent/SetParentTest.java]: is that used? I cannot found usage

**Comment by pvojtechovsky** [comment]
It looks good for me. Are you finished? Can I merge it now or will we wait 24h?

**Comment by monperrus** [comment]
We can merge now, since it does not change the code and only adds one 
new test.

We typically wait at least one day for major behavioral changes or new 
features.

**Comment by pvojtechovsky** [comment]
Thank You Martin!

**Pull request merged**